libraries/core_libs/consensus/src/transaction/transaction_queue.cpp

```diff
@@ -52,18 +73,50 @@ bool TransactionQueue::erase(const trx_hash_t &hash) {
   return true;
 }
 
-bool TransactionQueue::insert(std::shared_ptr<Transaction> &&transaction) {
-  assert(transaction);
-  if (hash_queue_.contains(transaction->getHash())) return false;
+bool TransactionQueue::insert(std::pair<std::shared_ptr<Transaction>, TransactionStatus> &&transaction,
+                              uint64_t last_block_number) {
+  assert(transaction.first);
+  if (hash_queue_.contains(transaction.first->getHash())) return false;
 
-  const auto tx_hash = transaction->getHash();
-  const auto it = priority_queue_.insert(std::move(transaction));
+  const auto tx_hash = transaction.first->getHash();
+  switch (transaction.second) {
+    case TransactionStatus::Verified: {
+      const auto it = priority_queue_.insert(std::move(transaction.first));
 
-  // This assert is here to check if priorityComparator works correctly. If object is not inserted, then there could be
-  // something wrong with comparator
-  assert(it != priority_queue_.end());
-  hash_queue_[tx_hash] = it;
+      // This assert is here to check if priorityComparator works correctly. If object is not inserted, then there could
+      // be something wrong with comparator
+      assert(it != priority_queue_.end());
+      hash_queue_[tx_hash] = it;
+    } break;
+    case TransactionStatus::LowNonce:
+      if (low_nonce_transactions_.size() < kLowNonceTransactionsLimit)
+        low_nonce_transactions_[transaction.first->getHash()] = {last_block_number, transaction.first};
+      break;
+    case TransactionStatus::InsufficentBalance:
+      if (insufficient_balance_transactions_.size() < kInsufficientBalanceTransactionsLimit)
+        insufficient_balance_transactions_[transaction.first->getHash()] = {last_block_number, transaction.first};
+      break;
+    default:
+      assert(false);
+  }
   return true;
 }
 
+void TransactionQueue::blockFinalized(uint64_t block_number) {
+  for (auto it = insufficient_balance_transactions_.begin(); it != insufficient_balance_transactions_.end();) {
+    if (it->second.first + kTransactionsExpiryLimit < block_number) {
+      it = insufficient_balance_transactions_.erase(it);
+    } else {
+      ++it;
+    }
+  }
+
+  for (auto it = low_nonce_transactions_.begin(); it != low_nonce_transactions_.end();) {
+    if (it->second.first + kTransactionsExpiryLimit < block_number)
```

Comment:
in the same loop above you use parenthesis, here you dont - inconsistent

Reply:
Fixed